[Shen Jiaming] iP

Oshen27 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Oshen27 committed:
Rename, Greet, Exit

Oshen27 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

Oshen27 committed:
Echo and Add List

Oshen27 committed:
Implemented Mark As Done feature

Added Task Class

Oshen27 committed:
Tweaked code to comply with coding standard

DanLinhHuynh-Niwashi commented on src/main/java/Bron.java:
```diff
@@ -0,0 +1,54 @@
+
+public class Bron {
+    public static void main(String[] args) {
+        String logo = "  ____    ____      ____    _     _\n"
+                    + " |  _ \\  |   _\\   /  __  \\ | \\   | |\n"
+                    + " | |_) | | |_) |  | |  | | |  \\  | | \n"
+                    + " |  _ <  | ___/   | |  | | | |\\\\ | |\n"
+                    + " | |_) | | | \\ \\  | |__| | | | \\\\| |\n"
+                    + " |____/  |_|  \\_\\  \\____/  |_|  \\__|\n";
```
Should this be a static variable?

Hackin7 replied:
Should `logo` be capitalised as `LOGO` instead to follow the naming convention for constants?

DanLinhHuynh-Niwashi commented on src/main/java/Bron.java:
```diff
@@ -0,0 +1,54 @@
+                    + " | |_) | | | \\ \\  | |__| | | | \\\\| |\n"
+                    + " |____/  |_|  \\_\\  \\____/  |_|  \\__|\n";
+        Scanner input = new Scanner(System.in);
+        System.out.println(logo + "Hello! I'm Bron\n" + "What can I do for you?\n");
```
You may change your name into a variable for easier reuse & modification (if any)

DanLinhHuynh-Niwashi commented on src/main/java/Bron.java:
```diff
@@ -0,0 +1,54 @@
+        while (true) {
+            String line = input.nextLine();
+
+            if (line.equalsIgnoreCase("bye")) {
+                System.out.println("Catch you on the flip cuh");
+                break;
+            }
+
+            if (line.equalsIgnoreCase("list")) {
+                int listCount = 0;
+                while(tasks[listCount] != null) {
+                    System.out.println(listCount + 1 + ". " + tasks[listCount++].printTask());
+                }
+            } else if (line.startsWith("mark")){
+                int taskIndex = Integer.parseInt(line.split(" ")[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskCount) {
+                    tasks[taskIndex].markAsDone();
+                    System.out.println("Good shit kid! I've marked this task as done:");
+                    System.out.println("  " + tasks[taskIndex].printTask());
+                } else {
+                    System.out.println("Task not found.");
+                }
+            } else if (line.startsWith("unmark")) {
+                int taskIndex = Integer.parseInt(line.split(" ")[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskCount) {
+                    tasks[taskIndex].markAsNotDone();
+                    System.out.println("Get yo shit together son, this task aint done yet:");
+                    System.out.println("  " + tasks[taskIndex].printTask());
+                } else {
+                    System.out.println("Task not found.");
+                }
+            } else {
+                System.out.println("added" + ": " + line);
+                tasks[taskCount++] = new Task(line);
+            }
```
You may separate the process of each command into methods for readability and reusability.

Should the command words (ex: "mark", "list", ...) be static variables?

DanLinhHuynh-Niwashi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = false;
+    }
+
+    public String printTask() {
+        return "[" + getStatusIcon() + "] " + this.description;
+    }
```
As the method is returning a String, not printing it out, should its name be changed to getTaskString() or something like that?

Hackin7 commented on src/main/java/Bron.java:
```diff
@@ -0,0 +1,54 @@
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        while (true) {
```
Should the main function be shortened (eg. with more abstraction)? This `main` function is quite hard to read for me as it is quite long.

Hackin7 commented on text-ui-test/runtest.sh:
Should there be something in this file to prevent people from being mislead, or at least, a TODO in this file to understand what needs to be done?

Hackin7 commented on src/main/java/Bron.java:
```diff
@@ -0,0 +1,54 @@
+                while(tasks[listCount] != null) {
+                    System.out.println(listCount + 1 + ". " + tasks[listCount++].printTask());
+                }
+            } else if (line.startsWith("mark")){
```
This looks like arrowhead code, and I feel with more abstraction we could reduce this effect

Oshen27 committed:
Added Todo, Deadline and Event classes

Oshen27 committed:
Implemented switch case for main loop, updated main loop with commands for adding Todo, deadline and event

Oshen27 committed:
Refactored code to improve code quality

Created Command enum, CommandHandler class and ChatBotManager class to improve readability and organization of code

Oshen27 committed:
Implemented automated text UI testing

Oshen27 committed:
Added Exception Handling to Bron

Added new Exception classes and updated error handling mechanism in CommandHandler

Oshen27 committed:
Added Exception Handling to Bron

Added new Exception classes and updated error handling mechanism in CommandHandler

Oshen27 committed:
Revert "Added Exception Handling to Bron"

This reverts commit 03196e64c6cbb603b9cbd5c24f17b239c45d078f.

Oshen27 committed:
Updated ChatBotManager to remove redundant code

Oshen27 committed:
Organized the classes into suitable packages

Oshen27 committed:
Updated runtest.sh for accurate testing

dZhei98 commented on src/main/java/bron/Bron.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    private static Task[] initializeTasks() {
+        return new Task[100];
```
Avoid the use of magic numbers.

Oshen27 committed:
Changed data structure for storing tasks to ArrayList

Oshen27 committed:
Implemented delete command

Oshen27 committed:
Implemented TaskStorage class for saving and writing to files

Oshen27 committed:
Merge branch 'branch-Level-7' into branch-Level-6

Oshen27 committed:
Updated task classes to include toSaveFormat() method

Oshen27 committed:
Fixed delete command and improved error handling

Oshen27 committed:
Implemented file saving to hard drive

Oshen27 committed:
Updated task constructors.
Implemented reading of data from hard drive upon chatbot launch

Oshen27 committed:
Updated error handling and implemented jar file

Oshen27 committed:
Updated task saving method

Oshen27 committed:
Implemented TaskList class

Oshen27 committed:
Implemented TextUI class to handle all interactions with user

Oshen27 committed:
Implemented Parser class

Oshen27 committed:
Updated deadline class to work with LocalDateTime

Oshen27 committed:
Implemented Find by keyword Command

Oshen27 committed:
Merge pull request #1 from Oshen27/branch-Level-8

Branch level 8

Oshen27 committed:
Merge branch 'branch-Level-9'

Oshen27 committed:
Merge pull request #2 from Oshen27/branch-Level-9

Branch level 9

Oshen27 committed:
Merge branch 'master' of https://github.com/Oshen27/ip

Oshen27 committed:
Implemented java doc for task package

Oshen27 committed:
Implemented JavaDoc for remaining classes

Oshen27 committed:
Implemented Help command for users to view how to use the programme

Oshen27 committed:
Added JavaDoc for Parser

Oshen27 committed:
Updated UserGuide with first draft

Oshen27 committed:
Updated UserGuide

Oshen27 committed:
Updated readme

Oshen27 committed:
Updated javaDoc
